var/spack/repos/builtin/packages/sst-macro/package.py

```diff
@@ -16,69 +16,92 @@ class SstMacro(AutotoolsPackage):
+    depends_on('m4', type='build', when='@master:')
 
     depends_on('binutils', type='build')
     depends_on('zlib', type=('build', 'link'))
     depends_on('otf2', when='+otf2')
-    depends_on('llvm+clang@:5.99.99', when='+skeletonizer')
-    depends_on('mpi', when='+mpi')
-    depends_on('sst-core@8.0.0', when='@8.0.0 +core')
-    depends_on('sst-core@develop', when='@develop +core')
-
-    variant('core', default=False, description='Use SST Core for PDES')
-    variant('mpi', default=True, description='Enable distributed PDES simulation')
-    variant('otf2', default=False, description='Enable OTF2 trace emission and replay support')
-    variant('shared', default=True, description='Build shared libraries')
-    variant('skeletonizer', default=False, description='Enable Clang source-to-source autoskeletonization')
+    depends_on('llvm+clang@5:9', when='+skeletonizer')
+    depends_on('mpi', when='+pdes_mpi')
+    depends_on('sst-core@devel',   when='@devel+core')
+    depends_on('sst-core@master',  when='@master+core')
+
+    variant('pdes_threads', default=True,
+            description='Enable thread-parallel PDES simulation')
+    variant('pdes_mpi', default=False,
+            description='Enable distributed PDES simulation')
+    variant('core',     default=False, description='Use SST Core for PDES')
+    variant('otf2',     default=False,
+            description='Enable OTF2 trace emission and replay support')
+    variant('skeletonizer', default=False,
+            description='Enable Clang source-to-source autoskeletonization')
+
     variant('static', default=True, description='Build static libraries')
-    variant('threaded', default=False, description='Enable thread-parallel PDES simulation')
+    variant('shared', default=True, description='Build shared libraries')
+
+    variant('werror', default=False,
+            description='Build with all warnings as errors')
+    variant('warnings', default=False,
+            description='Build with all possible warnings')
 
-    @run_before('autoreconf')
-    def bootstrap(self):
-        if '@develop' in self.spec:
-            Executable('./bootstrap.sh')()
+    # force out-of-source builds
+    build_directory = 'spack-build'
+
+    def autoreconf(self, spec, prefix):
+        bash = which('bash')
+        bash('./bootstrap.sh')
 
     def configure_args(self):
         args = ['--disable-regex']
 
-        # Set CFLAGS and CXXFLAGS so they won't automatically insert '-g'
-        env['CFLAGS'] = '-O2'
-        env['CXXFLAGS'] = '-O2'
-
         spec = self.spec
         args.append(
             '--enable-static=%s' % ('yes' if '+static' in spec else 'no'))
         args.append(
             '--enable-shared=%s' % ('yes' if '+shared' in spec else 'no'))
 
-        if spec.satisfies("@8.0.0:"):
+        if (spec.satisfies("@8.0.0:")
+            or spec.satisfies("@devel")
+                or spec.satisfies("@master")):
```

Comment:
Both `develop` and `master` satisfy `@8.0.0:` so you can remove those two from this if-statement.